### PR TITLE
SUBMARINE-221. Update submarine version to 0.3.0-SNAPSHOT in mini-submarine.

### PR DESCRIPTION
**Pull request by yuanzac**
### What is this PR for?
The way to use submarine is quite different from the version 0.2.0. We need to update mini-submarine with the latest code so that users can try submarine more conveniently.

### What type of PR is it?
Improvement

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-221

### How should this be tested?
https://travis-ci.org/yuanzac/hadoop-submarine/builds/595068622

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? Yes
* Does this needs documentation? Yes

### CODE REVIEW

File: dev-support/mini-submarine/conf/submarine.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
 <configuration>
   <property>
     <name>submarine.runtime.class</name>
-    <value>org.apache.hadoop.yarn.submarine.runtimes.tony.TonyRuntimeFactory</value>
+    <value>org.apache.submarine.runtimes.tony.TonyRuntimeFactory</value>
     <!--
     <value>org.apache.hadoop.yarn.submarine.runtimes.yarnservice.YarnServiceRuntimeFactory</value>
     -->
```
Comment (pingsutw) [inline dev-support/mini-submarine/conf/submarine.xml]: we also need to change it to **org.apache.submarine.runtimes.yarnservice.YarnServiceRuntimeFactory**

File: pom.xml [hunk 2]
```diff
@@ -185,6 +185,11 @@
         <artifactId>jackson-annotations</artifactId>
         <version>${jackson-annotations.version}</version>
       </dependency>
+      <dependency>
+        <groupId>com.fasterxml.jackson.core</groupId>
```
Comment (liuxunorg) [inline pom.xml]: Why do you need to import `jackson` jar?

File: submarine-core/pom.xml [hunk 3]
```diff
@@ -124,6 +124,10 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
```
Comment (liuxunorg) [inline submarine-core/pom.xml]: Why do you need to import `jackson` jar?

File: dev-support/mini-submarine/conf/submarine.xml (hunk 1 above)
Comment (yuanzac) [reply dev-support/mini-submarine/conf/submarine.xml]: It makes sense to me.
Thanks, @pingsutw

File: pom.xml (hunk 2 above)
Comment (yuanzac) [reply pom.xml]: I just found a bug that, if we don't import jackson jar, we would get the error that jackson class is not found, when using submarine-all-*.jar to submit a submarine job.

File: submarine-core/pom.xml (hunk 3 above)
Comment (yuanzac) [reply submarine-core/pom.xml]: The error that jackson class is not found if jackson-core is not imported

**Comment by liuxunorg** [comment]
@yuanzac , PR title loss word, `in mini-sub…`

**Comment by yuanzac** [comment]
> @yuanzac , PR title loss word, `in mini-sub…`

The title of this PR is changed

**Comment by yuanzac** [comment]
I'll merge this pr later today, if there are no other comments.

**Pull request closed without merging**